pkg/skaffold/build/cluster/cluster.go

```diff
@@ -60,12 +62,19 @@ func (b *Builder) buildArtifact(ctx context.Context, out io.Writer, artifact *la
 }
 
 func (b *Builder) runBuildForArtifact(ctx context.Context, out io.Writer, a *latest.Artifact, tag string) (string, error) {
+	// required artifacts as build-args
+	deps := docker.CreateBuildArgsFromArtifacts(a.Dependencies, b.artifactStore, true)
 	switch {
 	case a.KanikoArtifact != nil:
+		buildArgs, err := docker.EvalBuildArgs(b.mode, a.Workspace, a.KanikoArtifact.DockerfilePath, a.KanikoArtifact.BuildArgs, deps)
+		if err != nil {
+			return "", fmt.Errorf("unable to evaluate build args: %w", err)
+		}
+		a.KanikoArtifact.BuildArgs = buildArgs
 		return b.buildWithKaniko(ctx, out, a.Workspace, a.KanikoArtifact, tag)
 
 	case a.CustomArtifact != nil:
-		return custom.NewArtifactBuilder(nil, b.cfg, true, b.retrieveExtraEnv()).Build(ctx, out, a, tag)
+		return custom.NewArtifactBuilder(nil, b.cfg, true, append(b.retrieveExtraEnv(), util.EnvPtrMapToSlice(deps, "=")...)).Build(ctx, out, a, tag)
```

Comment:
We're using `deps` here which is now purported to be specific to docker build-args.  Rather than move `CreateBuildArgsFromArtifacts` to the `docker` package, could we rename it to something more general purpose, like `ResolveDependencyAliases()`?

Reply:
I can rename it but I'd want it to be in the `docker` package since I need it out of the `build` package to avoid cyclic package references and `docker` seemed like a good place.

Reply:
Argh!